src/bms/error.rs

```diff
@@ -5,15 +5,18 @@ use std::ops::RangeInclusive;
 use num::BigUint;
 use thiserror::Error;
 
-use super::{command::ObjId, prelude::*};
+use super::{
+    command::{ObjId, channel::ChannelIdParseWarning},
+    prelude::*,
+};
 
 /// An error occurred when parsing the [`TokenStream`].
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Error)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub enum ParseError {
+pub enum ControlFlowWarning {
```

Comment:
Problems on control flow tokens could ruin the semantics of source, so I marked it as an error but not a warning.
For unfolding that the library detected unsoundness, how about `ControlFlowError`?